algorithms/alg.verify-tx.tex

```diff
@@ -0,0 +1,19 @@
+\begin{algorithm}[H]
+    \caption{\label{alg.verify-tx} A function to verify that a transaction is conforming to the provided event and is included in the provided block.}
+    \begin{algorithmic}[1]
+        \Function{\sf verify-tx$_e$}{$\alpha$}
+            \Let{tx}{\textsf{decodeRawTX}(\alpha.\textsf{txRaw})}
+            \Let{blk}{\textsf{decodeBlockHeader}(\alpha.\textsf{blockHeader})}
+            \If{$\neg \textsf{containsTransfer}(tx, e.\textsf{receivingAddress}, e.\textsf{amount})$}
+                \State\Return{$\false$}
+            \EndIf
+            \Let{\textsf{txMerkleRoot}}{\textsf{calculateMerkleRoot}(H(\alpha.\textsf{txRaw}), \alpha.\textsf{txInclusion})}
```

Comment:
You need to explain what \calculateMerkleRoot is in the text. Also define it as a macro.

Reply:
Similarly H needs explaining.